include/concurrencpp/platform_defs.h

```diff
@@ -25,6 +25,28 @@
 #    define CRCPP_DEBUG_MODE
 #endif
 
+#if defined(CRCPP_WIN_OS)
+#    if defined(CRCPP_EXPORT_API)
+#        if __has_cpp_attribute(gnu::dllexport)
+#            define CRCPP_API __attribute__((dllexport))
+#        else
+#            define CRCPP_API __declspec(dllexport)
+#        endif
+#    elif defined(CRCPP_IMPORT_API)
+#        if __has_cpp_attribute(gnu::dllimport)
+#            define CRCPP_API __attribute__((dllimport))
+#        else
+#            define CRCPP_API __declspec(dllimport)
+#        endif
+#    endif
+#elif (defined(CRCPP_EXPORT_API) || defined(CRCPP_IMPORT_API)) && __has_cpp_attribute(gnu::visibility)
```

Comment:
So originally I was using `[[gnu::visibility("default")]]` rather than `__attribute__((visibility("default")))`, and testing for it is fine per the standard: https://en.cppreference.com/w/cpp/feature_test. The reason why I switched to the later is due to a current oversight with C++ attributes, see https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0537r0.html.

Reply:
@friendlyanon @ohanar 
please come to a conclusion on this

Reply:
This code this was a comment on was removed in c572788fcc8e101744855007f16cc001fc2b9b8e but this was not marked resolved.

Reply:
I didn't mark this one as resolved, because https://github.com/David-Haim/concurrencpp/commit/c572788fcc8e101744855007f16cc001fc2b9b8e addressed the conversation right above this one. @friendlyanon Did you have any more comments on `gnu::visibility`? (Not on `gnu::dll**port`.)

Reply:
Not familiar with using GCC [[attributes]] on Windows. My main goal was to inquire here.

Reply:
Okay, I'll mark this as resolved then, since `#elif` only occurs on non-Windows platform.